Addons/NetworkZones/NetworkZones_MySql.cs

```diff
@@ -0,0 +1,107 @@
+
+using UnityEngine;
+using UnityEngine.Networking;
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections.Generic;
+using System.Data;
+using MySql.Data;								// From MySql.Data.dll in Plugins folder
+using MySql.Data.MySqlClient;                   // From MySql.Data.dll in Plugins folder
+
+
+using SqlParameter = MySql.Data.MySqlClient.MySqlParameter;
+
+
+/// <summary>
+/// Database class for mysql
+/// Port of the sqlite database class from ummorpg
+/// </summary>
+public partial class Database
+{
+
+    static void Initialize_Zone()
+    {
+        ExecuteNonQueryMySql(@"
+        CREATE TABLE IF NOT EXISTS character_scene (
+            `character` VARCHAR(16) NOT NULL, 
+            scenepath VARCHAR(255) NOT NULL,
+            PRIMARY KEY(`character`)
+            ) CHARACTER SET=utf8mb4");
+
+
+        ExecuteNonQueryMySql(@"
+        CREATE TABLE IF NOT EXISTS zones_online (
+            online VARCHAR(255) NOT NULL
```

Comment:
Change this to TIMESTAMP

Reply:
this is the zone online check to stay consistend with the addon i let it as VARCHAR(). But you point me in the right direction here i have to change it
```#c
public static bool IsCharacterOnlineAnywhere(string characterName)
    {
        object obj = ExecuteScalarMySql("SELECT online FROM characters WHERE name=@name", new SqlParameter("@name", characterName));
        if (obj != null)
        {
            string online = (string)obj;
            if (online != "")
            {
                DateTime time = DateTime.Parse(online);
                double elapsedSeconds = (DateTime.UtcNow - time).TotalSeconds;
                float saveInterval = ((NetworkManagerMMO)NetworkManager.singleton).saveInterval;
                //UnityEngine.Debug.Log("datetime=" + time + " elapsed=" + elapsedSeconds + " saveinterval=" + saveInterval);
                return elapsedSeconds < saveInterval * 2;
            }
        }
        return false;
    }
```

Reply:
I suggested to vis to change this,  he will "look into it".   But even if he does not change it, I don't care about doing it exactly the same way vis did it.  I would rather do it right,  plus it is not like the fields need to maintain compatibility,  the method signatures are the ones that matter.